### PR TITLE
Add sensuctl describe-type command

### LINKED ISSUE #2376: Document the new "sensuctl describe-type" command
### Description
We will release a new `sensuctl describe-type` command in 5.20:

```bash
$ sensuctl describe-type -h
sensuctl describe-type

Describe a type by its fully qualified name:
$ sensuctl describe-type core/v2.CheckConfig

The command also supports describing types by their short names (for core/v2 types):
$ sensuctl describe-type checks,handlers

You can also use the 'all' qualifier to describe all available types:
$ sensuctl describe-type all

Usage:	sensuctl describe-type [RESOURCE TYPE],[RESOURCE TYPE]... [flags]

Flags:
      --format string   format of data returned ("wrapped-json"|"yaml") (default "none")
  -h, --help            help for describe-type

[...]

$ sensuctl describe-type all
      Fully Qualified Name           Short Name           API Version             Type          Namespaced
 ────────────────────────────── ───────────────────── ─────────────────── ──────────────────── ────────────
  authentication/v2.Provider                           authentication/v2   Provider             false
  licensing/v2.LicenseFile                             licensing/v2        LicenseFile          false
  store/v1.PostgresConfig                              store/v1            PostgresConfig       false
  federation/v1.EtcdReplicator                         federation/v1       EtcdReplicator       false
  secrets/v1.Secret                                    secrets/v1          Secret               true
  secrets/v1.Provider                                  secrets/v1          Provider             false
  searches/v1.Search                                   searches/v1         Search               true
  web/v1.GlobalConfig                                  web/v1              GlobalConfig         false
  core/v2.Namespace              namespaces            core/v2             Namespace            false
  core/v2.ClusterRole            clusterroles          core/v2             ClusterRole          false
  core/v2.ClusterRoleBinding     clusterrolebindings   core/v2             ClusterRoleBinding   false
  core/v2.User                   users                 core/v2             User                 false
  core/v2.APIKey                 apikeys               core/v2             APIKey               false
  core/v2.TessenConfig           tessen                core/v2             TessenConfig         false
  core/v2.Asset                  assets                core/v2             Asset                true
  core/v2.CheckConfig            checks                core/v2             CheckConfig          true
  core/v2.Entity                 entities              core/v2             Entity               true
  core/v2.Event                  events                core/v2             Event                true
  core/v2.EventFilter            filters               core/v2             EventFilter          true
  core/v2.Handler                handlers              core/v2             Handler              true
  core/v2.Hook                   hooks                 core/v2             Hook                 true
  core/v2.Mutator                mutators              core/v2             Mutator              true
  core/v2.Role                   roles                 core/v2             Role                 true
  core/v2.RoleBinding            rolebindings          core/v2             RoleBinding          true
  core/v2.Silenced               silenced              core/v2             Silenced             true
```

This will deprecate the `sensuctl dump --types` flag. See the PR: https://github.com/sensu/sensu-go/pull/3680

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [ ] update (add missing or refresh existing content)
- [x] improvement (improve docs content or format)

### Affected pages

I'm wondering how we should deal with resources listing in our docs. This new command will properly document the possible types for sensuctl [edit|dump|prune] but misses the RBAC names for enterprise resources (this is expected).

- https://docs.sensu.io/sensu-go/latest/sensuctl/reference/#sensuctl-edit-resource-types
- https://docs.sensu.io/sensu-go/latest/reference/rbac/#resources

### Additional context

* Browser(s) and version(s): 
* OS:

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
Adds `sensuctl describe-type` section in sensuctl reference doc.

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2376

### CODE REVIEW

**Comment by hillaryfraley** [comment]
I wasn't able to test this command. I am using an up-to-date local version (devel), but here's what I see when I try to list checks:
```
sensuctl describe-type core/v2.CheckConfig
Error: unknown command "describe-type" for "sensuctl"
```

**Comment by hillaryfraley** [comment]
I'm not sure what to add in the RBAC doc. Is there another flag for specifying a namespace to restrict the list to?

**Comment by hillaryfraley** [comment]
@palourde I am still getting `Error: unknown command "describe-type" for "sensuctl"`
I went ahead with a draft based on your notes in the issue. I'm not sure what to add in the RBAC doc. Is there another flag for specifying a namespace to restrict the list to?

**Comment by palourde** [comment]
@hillaryfraley For the RBAC page, we are currently missing documentation on a few types:

- `provider` (Postgres Store Provider, cluster-wide)
- `secrets` (Secrets, cluster-wide)
- `providers` (Secrets Providers, cluster-wide)
- `searches` (Web UI searches queries, namespaced)

There's no namespace flag support for this command.

File: content/sensu-go/5.20/sensuctl/reference.md [hunk 1]
```diff
@@ -468,15 +468,34 @@ To export only handlers and filters to a file named `my-handlers-and-filters.yam
...
-You can use the sensuctl `--types` subcommand to list the types of supported resources:
+{{% notice important %}}
+**IMPORTANT**: The `sensuctl describe-type` command deprecates `sensuctl dump --types`.
+{{% /notice %}}
+
+Use `sensuctl describe-type` to list the types of supported resources.
+For example, to list all types:
+
+{{< highlight shell >}}
+sensuctl describe-type --all
```
Comment (palourde) [inline content/sensu-go/5.20/sensuctl/reference.md]: `all` is actually an argument and not a flag, so it's going to be `sensuctl describe-type all`.

Comment (hillaryfraley) [inline content/sensu-go/5.20/sensuctl/reference.md]:
```suggestion
sensuctl describe-type all
```

**Review by palourde — APPROVED** [review]
🎉

**Pull request merged**